app/controllers/membership_applications_controller.rb

```diff
@@ -48,6 +48,12 @@ def create
 
       if new_file_uploaded(params)
         helpers.flash_message(:notice, t('.success'))
+
```

Comment:
This logic shouldn't go under the "if there is a new file uploaded..." condition.  It should be it's own small, simple method, and should be called after this conditional structure.

As Robert noted, once #318  is accepted, then you can use the `ShfMailer.`  Let `ShfMailer` have the responsibility for when and how to deliver: define a method in ShfMailer to send the mail, and put all of the logic for sending it in that method.  (Don't use `deliver_later` in this controller; keep the responsibility within the mailer used))

Reply:
@sic-f  Now this seems to be the case. =)

Reply:
@sic-f -- Now this is ready to be worked on. The PR that was actually merged was #359 .

 You should probably close this one and open a new PR since so much of the code base has changed.  You will also need to:
- get the exact text and formatting needed for the emails 

Doing this should now be really easy, based on the work for #359.

Reply:
@thesuss @weedySeaDragon I have reworked this PR. Ready to be reviewed again.